cdap-client/src/main/java/co/cask/cdap/client/MetricsClient.java

```diff
@@ -59,46 +62,92 @@ public MetricsClient(ClientConfig config) {
-                                                context == null ? "" : "&context=" + contextToPathParam(context),
-                                                groupBy == null ? "" : "&groupBy=" + groupBy));
+    List<String> queryParts = Lists.newArrayList();
+    queryParts.add("target=tag");
+    for (String tag : tags) {
+      queryParts.add("tag=" + tag);
+    }
+
+    URL url = config.resolveURLV3(String.format("metrics/search?%s", Joiner.on("&").join(queryParts)));
     HttpResponse response = restClient.execute(HttpMethod.POST, url, config.getAccessToken());
-    return ObjectResponse.fromJsonBody(response, MetricQueryResult.class).getResponseObject();
+    ObjectResponse<List<MetricTagValue>> result = ObjectResponse.fromJsonBody(
+      response, new TypeToken<List<MetricTagValue>>() { }.getType());
+    return result.getResponseObject();
   }
 
-  public MetricQueryResult query(String metric, Map<String, String> context)
+  /**
+   * Searches for metrics matching the given tags.
+   *
+   * @param tags the tags to match
+   * @return the metrics matching the given tags
+   * @throws IOException if a network error occurred
+   * @throws UnauthorizedException if the request is not authorized successfully in the gateway server
+   */
+  public List<String> searchMetrics(List<String> tags)
     throws IOException, UnauthorizedException {
-    return query(metric, context, null, null, null);
+
+    List<String> queryParts = Lists.newArrayList();
+    queryParts.add("target=metric");
+    for (String tag : tags) {
+      queryParts.add("tag=" + tag);
+    }
+
+    URL url = config.resolveURLV3(String.format("metrics/search?%s", Joiner.on("&").join(queryParts)));
+    HttpResponse response = restClient.execute(HttpMethod.POST, url, config.getAccessToken());
+    ObjectResponse<List<String>> result = ObjectResponse.fromJsonBody(
+      response, new TypeToken<List<String>>() { }.getType());
+    return result.getResponseObject();
   }
 
-  public MetricQueryResult query(String metric) throws IOException, UnauthorizedException {
-    return query(metric, null, null, null, null);
+  /**
+   * Gets the value of the given metrics.
+   *
+   * @param metrics names of the metrics
+   * @param groupBys groupBys for the request
+   * @param tags tags for the request
+   * @return values of the metrics
+   * @throws IOException if a network error occurred
+   * @throws UnauthorizedException if the request is not authorized successfully in the gateway server
+   */
+  public MetricQueryResult query(List<String> metrics, List<String> groupBys, List<String> tags)
+    throws IOException, UnauthorizedException {
+
+    List<String> queryParts = Lists.newArrayList();
+    queryParts.add("target=tag");
+    for (String metric : metrics) {
+      queryParts.add("metric=" + metric);
+    }
+    for (String groupBy : groupBys) {
+      queryParts.add("groupBy=" + groupBy);
+    }
+    for (String tag : tags) {
+      queryParts.add("tag=" + tag);
+    }
```

Comment:
we should support time range queries too right ? if we have to add time series, have  options to add interpolate and resolution as well.